crates/forge/bin/cmd/test/mod.rs

```diff
@@ -307,12 +307,20 @@ impl TestArgs {
 
         let env = evm_opts.evm_env().await?;
 
+        // If we are provided with test function regex, we are enabling complete internal fns
+        // tracing.
+        let decode_internal = self.decode_internal.as_ref().map_or(false, |v| v.is_some());
+        // If we are provided with just --decode-internal flag, we enable simple tracing (without
+        // memory decoding).
+        let decode_internal_simple = self.decode_internal.is_some();
```

Comment:
Can we have an enum just for this? Full, Simple, No

Reply:
added in 92ece42